src/features/operator/Market/MarketWithdraw/index.tsx

```diff
@@ -118,6 +129,170 @@ export const MarketWithdraw = (): JSX.Element => {
     }
   };
 
+  const baseToPreferredCurrency = React.useMemo(() => {
+    // Wait for prices to load (or error out) before displaying conversions
+    if (isLoadingPrices || isErrorPrices) {
+      return '';
+    }
+
+    let amount = Big(0);
+    try {
+      amount = Big(balanceBaseAmount);
+    } catch {
+      // ignore user typos, just leave the amount as 0
+    }
+    const assetId = selectedMarket.baseAsset?.asset_id || '';
+    const assetName = selectedMarket.baseAsset?.ticker || 'unknown';
+
+    let amountInFiatOrLBTC = '';
+    if (isLbtcAssetId(selectedMarket.baseAsset?.asset_id || '', network)) {
+      amountInFiatOrLBTC = formatLbtcUnitToSats(amount.toNumber(), lbtcUnit);
+      amountInFiatOrLBTC = formatSatsToUnit(Number(amountInFiatOrLBTC), 'L-BTC', assetId, network);
+    } else {
+      amountInFiatOrLBTC = formatSatsToUnit(amount.times(100000000).toNumber(), 'L-BTC', assetId, network);
+    }
+
+    let rateMultiplier = 1;
+    let preferredCurrencyAmount = Big(1);
+    if (assetName.includes('L-BTC')) {
+      rateMultiplier = prices?.[LBTC_COINGECKOID]?.[currency.value] || 1;
+      preferredCurrencyAmount = Big(amountInFiatOrLBTC).times(rateMultiplier);
+    } else if (assetName === 'USDt') {
+      rateMultiplier = prices?.[USDT_COINGECKOID]?.[currency.value] || 1;
+      preferredCurrencyAmount = Big(amountInFiatOrLBTC).times(rateMultiplier);
+    } else if (assetName === 'LCAD') {
+      // COINGECKO does not have a ticker for LCAD
+      // We have to manually calculate the rates
+
+      // CAD -> BTC
+      const BTC_CAD_RATE = prices?.[LBTC_COINGECKOID]?.['cad'] || 1;
```

Comment:
I don't understand from where `prices?.[LBTC_COINGECKOID]?.['cad']` is coming

Reply:
The CoinGecko price feed has the following structure

prices['L-BTC']['usd']
prices['L-BTC']['eur']
prices['L-BTC']['cad']
prices['L-BTC']['btc']
prices['USDT']['usd']
prices['USDT']['eur']
prices['USDT']['cad']
prices['USDT']['btc']

Since there is no **prices['L-CAD']['btc']** structure we calculate the CAD_TO_BTC rate manually using the following pseudocode

```
BTC_CAD_RATE = prices['L-BTC']['cad']
CAD_BTC_RATE = 1/BTC_CAD_RATE
```

Reply:
I would move that CAD_TO_BTC calculation to `rates.api.ts`